### PR TITLE
strap.sh: implementing new errors check and refactoring.

### PR DESCRIPTION
**Pull request by PedroSFreitas**
We're seeing a lot of users on IRC (server: Freenode) receiving "invalid keyring" error message when it usually is a network/internet misconfiguration or problem.
This PR tries to create useful checks and error messages, although it's probably not 100% correct. It'd be interesting to have a second eyes on it.
Any critics, positive or negative are much appreciated.

And I tried to refactor a few things according to https://github.com/koalaman/shellcheck

Note: I've tried to provide enough information on the commit messages, it'd be interesting to read them in order to understand the changes.

### CODE REVIEW

**Comment by PedroSFreitas** [comment]
@noptrix A review would be great, if you have time. Thanks.

File: strap.sh [hunk 1]
```diff
@@ -50,15 +63,21 @@ fetch_keyring()
 # note: this is pointless if you do not verify the key fingerprint
 verify_keyring()
 {
-    gpg \
-        --keyserver pgp.mit.edu \
-        --recv-keys 4345771566D76038C7FEB43863EC0ADBEA87E4E3 > /dev/null 2>&1
+    if ! gpg --keyserver pgp.mit.edu \
```
Comment (ikstream) [inline strap.sh:66]: you used if; then notation in the rest of the file. Wouldn't it be good to keep it consistent

**Review by ikstream — CHANGES_REQUESTED** [review]
looks good to me, except for the one 'if; then' style change

File: strap.sh (hunk 1 above)
Comment (PedroSFreitas) [reply strap.sh:66]: You are correct. What I thought was: when the condition have multiple lines I think the `then` in a new line makes it easier to the eyes. What do you think?

Comment (ikstream) [reply strap.sh:66]: I prefer ```if;then``` over ```then``` in a newline, but for better readability i prefer a consistent style. 
How about using ```then``` in a new line file wide, as i see your point for multi-line conditionals?

Comment (PedroSFreitas) [reply strap.sh:66]: Perfect and I agree. I'll make the changes later. Thanks @ikstream

**Comment by PedroSFreitas** [comment]
@noptrix 
image
without internet connection (disconnected)

image
if an error occur trying to get the key from the keyserver.

image
working.

**Comment by noptrix** [comment]
thanks a lot. i will merge the pr prior the new ISO releases (~3 weeks)

**Pull request merged**